src/components/ShadowedScrollView.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import React from "react";
+import { StyleSheet, ScrollView, View, Animated } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+
+type Props = {
+  children: Array<Object>,
+  style: StyleObj
+};
+
+const shadowFadeDuration: number = 100;
+const maxScrollEventThrottle: number = 16;
+const shadowHeight: number = 15;
+
+class ShadowedScrollView extends React.PureComponent<Props> {
+  static defaultProps = {
+    children: [],
+    style: {}
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  topShadowOpacity: Object;
+  bottomShadowOpacity: Object;
+  isTopShadowPresent: boolean = false;
+  isBottomShadowPresent: boolean = true;
+  contentViewHeight: number;
+  scrollViewHeight: number;
+
+  fadeTopShadow = (toValue: number) =>
+    Animated.timing(this.topShadowOpacity, {
+      toValue,
+      duration: shadowFadeDuration,
+      useNativeDriver: true
+    }).start();
+
+  fadeBottomShadow = (toValue: number) =>
+    Animated.timing(this.bottomShadowOpacity, {
+      toValue,
+      duration: shadowFadeDuration,
+      useNativeDriver: true
+    }).start();
+
+  componentWillMount() {
+    this.topShadowOpacity = new Animated.Value(0);
+    this.bottomShadowOpacity = new Animated.Value(1);
+  }
+
+  handleScroll = (event: { nativeEvent: { contentOffset: { y: number } } }) => {
+    const maxScrollOffset = this.contentViewHeight - this.scrollViewHeight - 10;
+    const currentScrollOffset = event.nativeEvent.contentOffset.y;
+
+    if (this.isTopShadowPresent && currentScrollOffset <= 0) {
+      this.isTopShadowPresent = false;
```

Comment:
This flag is needed to avoid triggering the `fadeTopShadow` multiple times as it slows the animation.
Also it's not on the component state to avoid triggering the component render.